moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/jog_arm_data.h

```diff
@@ -79,6 +79,10 @@ struct JogArmShared
 
   // The transform from the MoveIt planning frame to robot_link_command_frame
   Eigen::Isometry3d tf_moveit_to_cmd_frame;
+
+  // True -> allow drift in this dimension. In the command frame. [x, y, z, roll, pitch, yaw]
+  std::atomic_bool drift_dimensions[6] = { ATOMIC_VAR_INIT(false), ATOMIC_VAR_INIT(false), ATOMIC_VAR_INIT(false),
```

Comment:
Atomic bools are useful for preventing race conditions for independent flags without needing mutex guards. However, the service enforces that all flags are updated together using a mutex lock while the function `removeDimension()` doesn't block the filter for read access. Question is: are the flags independent from each other or not? If they are independent, there is no need for a mutex lock. If they aren't, both read and write access should be locked, but then there is no need for atomic bools.

Reply:
I think you're on to something. I believe the mutex lock in `changeDriftDimensions()` is not necessary since the std::atomic bools are threadsafe without it.

Reply:
Or we could go back to mutex-locking everything, no std::atomic bools as it originally was before Dave's suggestion. I'd prefer to keep the std::atomics, no mutex lock

Reply:
done